modules/flake-parts/writers.nix

```diff
@@ -9,10 +9,25 @@
   in {
     options.writers = {
       writePureShellScript = lib.mkOption {
-        type = lib.types.functionTo lib.types.anything;
+        type = lib.types.functionTo (lib.types.functionTo lib.types.package);
```

Comment:
I've moved the documentation here, but probably the option declarations should be moved closer to the writers.

Sort-of like this:

```nix
options.writers = lib.mkOption { type = submodule ../writers.nix; }
```